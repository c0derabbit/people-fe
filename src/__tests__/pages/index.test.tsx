import { getPage } from 'next-page-tester'
import { render } from '../testUtils'

describe('Home page', () => {
  (global as any).fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve({ records: TEST_DATA }),
    })
  )

  it('matches snapshot', async () => {
    const { page } = await getPage({ route: '/' })
    const { asFragment } = render(page)

    expect(asFragment()).toMatchSnapshot()
  })
})

const TEST_DATA = [
  {
    id: 'one',
  },
  {
    id: 'two',
  },
]
