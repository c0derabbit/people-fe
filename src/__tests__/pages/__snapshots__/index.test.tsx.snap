// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="css-nil ub-bs_17bru36 ub-bg_11ev722 ub-pb_8px ub-pt_8px ub-pst_absolute ub-z-idx_2 ub-w_100vw ub-box-szg_border-box"
  >
    <div
      class="css-nil ub-just-cnt_space-between ub-algn-itms_center ub-dspl_flex ub-max-w_1200px ub-ml_auto ub-mr_auto ub-pl_16px ub-pr_16px ub-box-szg_border-box"
    >
      <a
        class="ub-mt_0px ub-mb_0px ub-fnt-sze_14px ub-f-wght_400 ub-txt-trns_uppercase ub-ln-ht_16px ub-ltr-spc_0-6px ub-fnt-fam_b77syt ub-color_234361 ub-txt-deco_none ub-box-szg_border-box"
        href="/"
        rel="noopener"
      >
        Peoplegraph
      </a>
      <button
        class="css-69cngj ub-fnt-fam_b77syt ub-mt_0px ub-fnt-sze_12px ub-f-wght_500 ub-ln-ht_32px ub-ltr-spc_0 ub-btrr_3px ub-bbrr_3px ub-btlr_3px ub-bblr_3px ub-pt_0px ub-pb_0px ub-pr_16px ub-pl_16px ub-ml_0px ub-mr_0px ub-mb_0px ub-h_32px ub-pst_relative ub-dspl_inline-flex ub-algn-itms_center ub-flx-wrap_nowrap ub-box-szg_border-box"
      >
        Log out
      </button>
    </div>
  </header>
  <main
    class="css-nil ub-pl_16px ub-pr_16px ub-pb_70px ub-pt_70px ub-max-w_1200px ub-ml_auto ub-mr_auto ub-box-szg_border-box"
  >
    <select>
      <option
        value="id"
      >
        id
      </option>
    </select>
    <label>
      <input
        type="checkbox"
      />
      reverse
    </label>
    <ul
      class="css-nil ub-pb_0px ub-pl_0px ub-pr_0px ub-pt_0px ub-ls_iu2jf4 ub-box-szg_border-box"
    >
      <li>
        X. Y.
      </li>
      <li>
        Y. Z.
      </li>
    </ul>
  </main>
</DocumentFragment>
`;
